xla/hlo/ir/hlo_computation.cc

```diff
@@ -1068,7 +1068,7 @@ StatusOr<HloInstruction*> HloComputation::CreateAsyncInstructions(
   async_start->CopyBackendConfigFrom(instruction);
   async_done->set_metadata(instruction->metadata());
   async_done->CopyBackendConfigFrom(instruction);
-  TF_RETURN_IF_ERROR(async_done->CopyAllControlDepsFrom(instruction));
+  TF_RETURN_IF_ERROR(async_start->CopyAllControlDepsFrom(instruction));
```

Comment:
I think the right thing to do here would be to put the control predecessors to async_start and control successors to async_done

Reply:
ah good point, forgot that we had control successors.
updated